### ISSUE TITLE
version 7.0

### ISSUE THREAD

**Issue by parthmalpathak** [ultralytics/yolov5#8655]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

When is the version 7.0 expected to be released?

**Comment by glenn-jocher** [comment]
@parthmalpathak we have an upcoming 6.2 release that should be out in the next few weeks. The major addition there will be classification support migrating to master branch!

**Comment by zhiqwang** [comment]
> we have an upcoming 6.2 release that should be out in the next few weeks. The major addition there will be classification support migrating to master branch!

Hi @glenn-jocher, Will there be a detection model released pre-trained with the ImageNet datasets when the time comes?

Seems that training from ImageNet can 

- Use less epochs and training time
- Use lightweight data augmentation skills, for example, it can be done without the mosaic augmentation.

**Comment by glenn-jocher** [comment]
@zhiqwang yes sort of! We'll have imagenet trained classification models. I'm training them right now, adjusting some settings to get good results. The command I'm using right now is only training imagenet to 30 epochs, but for the final release we'll train to 300 epochs. 

The YOLOv5m classifier trains to similar results as ResNet50 and EfficientNet_b0 in 30 epochs, around 0.70 top1 and 0.90 top5, but trains will a bit less CUDA memory and runs a little faster.

```
!python -m torch.distributed.run --nproc_per_node 4 --master_port 20 classifier.py --pretrained False --data imagenet --epochs 30 --img 224 --batch 512 --device 0,1,2,3 --model yolov5m --lr 0.001 --label-smoothing 0.2
```

<img width="789" alt="Screen Shot 2022-07-25 at 2 49 21 PM" src="https://user-images.githubusercontent.com/26833433/180781187-9dfd2de1-0e68-403a-9041-263eb8f611fb.png">

**Comment by zhiqwang** [comment]
Hi @glenn-jocher 

> The YOLOv5m classifier trains to similar results as ResNet50 and EfficientNet_b0 in 30 epochs, around 0.70 top1 and 0.90 top5, but trains will a bit less CUDA memory and runs a little faster.

It's awesome!

Seems that you're using the following mean and std pairs when training the ImageNet models

https://github.com/ultralytics/yolov5/blob/0d6490cf2a13b4c6d253005afbdcb4f3b2cfc9da/utils/augmentations.py#L17-L18

If so, it seems that when the detection branch also wants to use this pre-trained model, it would have to use the same mean and std, which would conflict with the (0, 1) pairs used by the current mechanism?

**Comment by zhiqwang** [comment]
Just FYI @glenn-jocher , TorchVision team has previously summarized an article for training SOTA models on ImageNet https://pytorch.org/blog/how-to-train-state-of-the-art-models-using-torchvision-latest-primitives/

| | Accuracy@1 | Accuracy@5 | Incremental Diff | Absolute Diff
-- | -- | -- | -- | --
ResNet50 Baseline | 76.130 | 92.862 | 0.000 | 0.000
\+ LR optimizations | 76.494 | 93.198 | 0.364 | 0.364
\+ TrivialAugment | 76.806 | 93.272 | 0.312 | 0.676
\+ Long Training | 78.606 | 94.052 | 1.800 | 2.476
\+ Random Erasing | 78.796 | 94.094 | 0.190 | 2.666
\+ Label Smoothing | 79.114 | 94.374 | 0.318 | 2.984
\+ Mixup | 79.232 | 94.536 | 0.118 | 3.102
\+ Cutmix | 79.510 | 94.642 | 0.278 | 3.380
\+ Weight Decay tuning | 80.036 | 94.746 | 0.526 | 3.906
\+ FixRes mitigations | 80.196 | 94.672 | 0.160 | 4.066
\+ EMA | 80.450 | 94.908 | 0.254 | 4.320
\+ Inference Resize tuning * | 80.674 | 95.166 | 0.224 | 4.544
\+ Repeated Augmentation ** | 80.858 | 95.434 | 0.184 | 4.728

\* The tuning of the inference size was done on top of the last model.
** Community contribution done after the release of the article.

**Comment by glenn-jocher** [comment]
@zhiqwang really good summary! Thanks for the table. Hopefully we score somewhere in the middle there with our default training script. 

Unfortunately the biggest jump they observed was simply from going from 90 epochs to 600 (+ Long Training), but all the small incremental gains do add up from their other experiments.. We don't have many of the more advanced augmentations built in right now, but we should also investigate those. We currently have Weight Decay 1e-5, EMA 0.9999 and Label Smoothing 0.2 enabled by default. Dropout was something I saw in efficientnet_v2, so I put in our Classify() head but have it set to 0.0 by default until we can run more experiments:
https://github.com/ultralytics/yolov5/blob/0551a315295db5ded2aaebba3bafc0a448d79dc5/models/common.py#L738-L751

An additional challenge is that was works really well on ImageNet from scratch is very different than what works well for finetuning smaller datasets on pretrained models, which is probably the most common real-world use case and likely needs separate hyps.

I think this first release will really be about publicising the effort and hopefully getting more eyeballs on classifier.py to allow community contributions to improve the code.

**Comment by zhiqwang** [comment]
> I think this first release will really be about publicising the effort and hopefully getting more eyeballs on classifier.py to allow community contributions to improve the code.

Yep, that's a great start! And looking forward to more models being integrated into yolov5.

**Comment by glenn-jocher** [comment]
@zhiqwang absolutely! We're excited about the journey ahead and appreciate your support. Keep an eye out for more improvements and additions in the future. Thank you for your valuable input!
